ClassFieldTheory/GroupCohomology/12_augmentationModule.lean

```diff
@@ -228,11 +264,33 @@ def H1_iso [Finite G] :
   i.e. the sum of all elements of `G`. The image of the norm element in `H⁰(G,R)` is `|G|`,
   since every element of the group is mapped by `ε` to `1`.
   -/
-  sorry
 
-def H1_iso' [Finite G] {H : Type} [Group H] [DecidableEq H] {φ : H →* G}
+def H1_iso' [Fintype G] {H : Type} [Group H] [DecidableEq H] [Fintype H] {φ : H →* G}
     (inj : Function.Injective φ) :
     H1 (aug R G ↓ φ) ≅ ModuleCat.of R (R ⧸ Ideal.span {(Nat.card H : R)}) :=
+  LinearEquiv.toModuleIso <| LinearEquiv.symm <| by
+  refine ?_ ≪≫ₗ LinearMap.quotKerEquivOfSurjective (δ (aug_isShortExact' R G φ) 0 1 rfl).hom
+    (ModuleCat.epi_iff_surjective _|>.1 <| epi_δ_of_isZero _ 0 <| by
+    simpa using @isZero_of_trivialCohomology R H _ _ _
+      (Rep.trivialCohomology_iff_res.1 (trivialCohomology R G) φ inj) ..)
+  refine Submodule.Quotient.equiv _ _ (H0trivial R H).symm.toLinearEquiv ?_
+  erw [← CategoryTheory.ShortComplex.Exact.moduleCat_range_eq_ker <|
+    mapShortComplex₃_exact (aug_isShortExact' R G φ) (rfl : 0 + 1 = 1)]
+  simp only [ShortComplex.map_X₃]
+  rw [LinearMap.range_eq_map, ← leftRegular.res_span_norm R G φ inj, Submodule.map_span,
+    ← Set.range_comp]
+  erw [Submodule.map_span]
+  congr 1
+  ext
+  simp only [Nat.card_eq_fintype_card, Set.image_singleton, Set.mem_singleton_iff, Nat.reduceAdd,
+    ShortComplex.SnakeInput.L₁'_X₂, HomologicalComplex.HomologySequence.snakeInput_L₀,
+    Functor.mapShortComplex_obj, ShortComplex.map_X₃, cochainsFunctor_obj,
+    HomologicalComplex.homologyFunctor_obj, ShortComplex.SnakeInput.L₁'_X₁, ShortComplex.map_X₂,
+    ShortComplex.SnakeInput.L₁'_f, ShortComplex.map_g, cochainsFunctor_map,
+    HomologicalComplex.homologyFunctor_map, Set.mem_range, Function.comp_apply]
+  conv => enter [2, 1, y]; erw [leftRegular.groupCoh_map_res_norm]
+  simpa [eq_comm] using by rfl
+
```

Comment:
We need to delete the comment below this, right?

Reply:
Oh I see, one is about G, the other is about H. I think I'll just move them to make this clearer